### PR TITLE
fix: limit 0 return empty

### PR DESCRIPTION
**Pull request by caicancai**

### CODE REVIEW

**Review by skyzh — COMMENTED** [review]
LGTM, but it seems that a lot of regress tests are failing because we don't populate data for these queries. Need to figure out a way to fix them before merging. Thanks anyways!

**Comment by caicancai** [comment]
> LGTM, but it seems that a lot of regress tests are failing because we don't populate data for these queries. Need to figure out a way to fix them before merging. Thanks anyways!

I'll try to fix it over the weekend.
Do we just need to fill in the data of the test sample? I am not very clear about this part. Can you give me some advice?

File: src/planner/rules/plan.rs [hunk 1]
```diff
@@ -18,7 +18,7 @@ pub fn always_better_rules() -> Vec<Rewrite> {
 
 #[rustfmt::skip]
 fn cancel_rules() -> Vec<Rewrite> { vec![
-    rw!("limit-null";       "(limit null 0 ?child)"     => "?child"),
+    rw!("limit-null";       "(limit null 0 ?child)"     => "(empty ?child)"),
```
Comment (wangrunji0408) [inline src/planner/rules/plan.rs:21]: The definition of the limit node is `(limit limit_num offset_num child)`.
So `(limit null 0 ?child)` means `LIMIT ALL OFFSET 0`, which is a dummy operation.
Actually this is not a bug. 😄

**Review by wangrunji0408 — APPROVED** [review]

File: src/planner/rules/plan.rs (hunk 1 above)
Comment (caicancai) [reply src/planner/rules/plan.rs:21]: Thanks for the reminder, but limit 0 actually returns a logical plan similar to
`LogicalValues(tuples=[[]]).`
 I don't know if risinglight has implemented it now. If not, I'd be happy to contribute.

**Pull request closed without merging**